tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -85,653 +80,761 @@
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        ), "Should be able to create a topic with colliding chars after deletion.");
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDeleteInternalTopic(String quorum) throws Exception {
-        // create the offset topic
-        TestUtils.createTopicWithAdmin(adminClient, Topic.GROUP_METADATA_TOPIC_NAME, scalaBrokers, scalaControllers, defaultNumPartitions, defaultReplicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        // Try to delete the Topic.GROUP_METADATA_TOPIC_NAME which is allowed by default.
-        // This is a difference between the new and the old command as the old one didn't allow internal topic deletion.
-        // If deleting internal topics is not desired, ACLS should be used to control it.
-        TopicCommand.TopicCommandOptions deleteOffsetTopicOpts =
-                buildTopicCommandOptionsWithBootstrap("--delete", "--topic", Topic.GROUP_METADATA_TOPIC_NAME);
-        String deleteOffsetTopicPath = kafka.zk.DeleteTopicsTopicZNode.path(Topic.GROUP_METADATA_TOPIC_NAME);
-        if (!isKRaftTest()) {
-            assertFalse(zkClient().pathExists(deleteOffsetTopicPath), "Delete path for topic shouldn't exist before deletion.");
+    @ClusterTemplate("generate1")
+    public void testDeleteInternalTopic() throws Exception {
+        try (Admin adminClient = clusterInstance.createAdminClient();
+             TopicCommand.TopicService topicService = new TopicCommand.TopicService(adminClient)) {
+
+            // create the offset topic
+            // In ZK mode, Topic.GROUP_METADATA_TOPIC_NAME exist when cluster is created.
+            if (clusterInstance.isKRaftTest()) {
+                adminClient.createTopics(Collections.singletonList(new NewTopic(Topic.GROUP_METADATA_TOPIC_NAME, defaultNumPartitions, defaultReplicationFactor)));
+                clusterInstance.waitForTopic(Topic.GROUP_METADATA_TOPIC_NAME, defaultNumPartitions);
+            }
+
+            // Try to delete the Topic.GROUP_METADATA_TOPIC_NAME which is allowed by default.
+            // This is a difference between the new and the old command as the old one didn't allow internal topic deletion.
+            // If deleting internal topics is not desired, ACLS should be used to control it.
+            TopicCommand.TopicCommandOptions deleteOffsetTopicOpts =
+                    buildTopicCommandOptionsWithBootstrap("--delete", "--topic", Topic.GROUP_METADATA_TOPIC_NAME);
+
+            topicService.deleteTopic(deleteOffsetTopicOpts);
+            TestUtils.waitForCondition(
+                    () -> adminClient.listTopics().listings().get().stream().noneMatch(topic -> topic.name().equals(Topic.GROUP_METADATA_TOPIC_NAME)),
+                    60000, "Delete topic fail in 60000 ms"
```

Comment:
I can see that `60000` is set as max time out in many places can we extract this into a contestant instead?